src/NHibernate.Test/BulkManipulation/BulkOperationCleanupActionFixture.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NHibernate.Action;
+using NHibernate.Engine;
+using NHibernate.Metadata;
+using NHibernate.Persister.Entity;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace NHibernate.Test.BulkManipulation
+{
+	[TestFixture]
+	public class BulkOperationCleanupActionFixture
+	{
+		private ISessionImplementor _session;
+		private ISessionFactoryImplementor _factory;
+		private IEntityPersister _persister;
+
+		[SetUp]
+		public void SetupTest()
+		{
+			_session = Substitute.For<ISessionImplementor>();
+			_factory = Substitute.For<ISessionFactoryImplementor>();
+			_persister = Substitute.For<IEntityPersister>();
+			_session.Factory.Returns(_factory);
+			_factory.GetAllClassMetadata().Returns(new Dictionary<string, IClassMetadata> { ["TestClass"] = null });
+			_factory.GetEntityPersister("TestClass").Returns(_persister);
+			_factory.GetCollectionRolesByEntityParticipant("TestClass").Returns(new HashSet<string>(new[] { "TestClass.Children" }));
+			_persister.QuerySpaces.Returns(new[] { "TestClass" });
+			_persister.EntityName.Returns("TestClass");
+		}
+
+		[TestCase("TestClass", true, 1, 1, 1)]
+		[TestCase("AnotherClass", true, 1, 0, 0)]
+		[TestCase("AnotherClass,TestClass", true, 2, 1, 1)]
+		[TestCase("TestClass", false, 1, 0, 1)]
+		[TestCase("", true, 1, 1, 1)]
+		[Test]
+		// 6.0 TODO: remove this ignore.
+		[Ignore("Must wait for the tested methods to be actually added to ISessionFactoryImplementor")]
+		public void AfterTransactionCompletionProcess_EvictsFromCache(string querySpaces, bool persisterHasCache, int expectedPropertySpaceLength, int expectedEntityEvictionCount, int expectedCollectionEvictionCount)
+		{
+			_persister.HasCache.Returns(persisterHasCache);
+
+			var target = new BulkOperationCleanupAction(_session, new HashSet<string>(querySpaces.Split(new []{','},StringSplitOptions.RemoveEmptyEntries)));
+
+			target.AfterTransactionCompletionProcess(true);
+
+			Assert.AreEqual(expectedPropertySpaceLength, target.PropertySpaces.Length);
+
+			if (expectedEntityEvictionCount > 0)
+			{
+				_factory.Received(1).EvictEntity(Arg.Is<IEnumerable<string>>(x => x.Count() == expectedEntityEvictionCount));
```

Comment:
This fails compiling with [Travis](https://travis-ci.org/nhibernate/nhibernate-core/jobs/383615826#L2046).

```
BulkManipulation/BulkOperationCleanupActionFixture.cs(54,38): error CS0012: The type 'Expression<>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Linq.Expressions, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/home/travis/build/nhibernate/nhibernate-core/src/NHibernate.Test/NHibernate.Test.csproj]
BulkManipulation/BulkOperationCleanupActionFixture.cs(54,66): error CS1660: Cannot convert lambda expression to type 'IEnumerable<string>' because it is not a delegate type [/home/travis/build/nhibernate/nhibernate-core/src/NHibernate.Test/NHibernate.Test.csproj]
```

I have suspected that a reference was missing for .Net Core target, but it does compile fine under Windows. (AppVeyor build is successful so far, it has ended its SQL-Server and PostgreSQL build as of now.)

Moreover [the package it asks to add](https://www.nuget.org/packages/System.Linq.Expressions) looks to me as an obsolete thing: it does still depend on `.NETCore 5.0`, the old naming before they decided to number .Net Core from 1.0.

So this looks as a linux/mono specific issue.